test/exq_test.exs

```diff
@@ -117,6 +117,30 @@ defmodule ExqTest do
     assert_received {:worked, 2}
     stop_process(sup)
   end
+  
+  test "register queue and run job" do
+    Process.register(self, :exqtest)
+    {:ok, sup} = Exq.start_link([name: :exq_t, port: 6555, namespace: "test", queues: ["q1"]])
+    {:ok, _} = Exq.subscribe(:exq_t, "q2", 10)
+    {:ok, _} = Exq.enqueue(:exq_t, "q1", ExqTest.PerformArgWorker, [1])
+    {:ok, _} = Exq.enqueue(:exq_t, "q2", ExqTest.PerformArgWorker, [2])
+    wait_long
+    assert_received {:worked, 1}
+    assert_received {:worked, 2}
+    stop_process(sup)
+  end
+  
+  test "unregister queue and run job" do
+    Process.register(self, :exqtest)
+    {:ok, sup} = Exq.start_link([name: :exq_t, port: 6555, namespace: "test", queues: ["q1","to_remove"]])
+    {:ok, _} = Exq.unsubscribe(:exq_t, "to_remove")
```

Comment:
I think I figured this out - looks like an actual issue with the way things were getting dequeued, see #103 . Can you rebase against master and try this again?